src/FirmwarePlugin/FirmwarePlugin.cc

```diff
@@ -567,3 +567,32 @@ Autotune* FirmwarePlugin::createAutotune(Vehicle *vehicle)
 {
     return new Autotune(vehicle);
 }
+
+void FirmwarePlugin::updateAvailableFlightModes(FlightModeList modeList)
+{
+    _availableFlightModeList.clear();
+
+    for(auto mode: modeList){
+        _updateModeMappings(mode);
+    }
+}
+
+void FirmwarePlugin::_setModeEnumToModeStringMapping(FlightModeCustomModeMap enumToString)
+{
+    _modeEnumToString = enumToString;
+}
+
+void FirmwarePlugin::_updateModeMappings(FirmwareFlightMode &mode){
+    // Get Presaved Mode String, if mode is not present then take Custom Mode Name
+
+    QString nModeName = mode.mode_name;
```

Comment:
I wonder if this should be the other way around. Naming coming from standard modes takes precedence. That naming is likely what is going to be in the docs for the firmware. If we use the internal QGC naming that make differ.

The benefit of using the internal QGC naming is that it will end up being translated.

Not quite sure what is good here. Did you have specific reason to do it this way?

Reply:
So if we take the flight mode name from the standard modes then the issue is first of casing, ie arducopter sending all caps. Even if I add `Qt:CaseInsensitive` in the string comparison, in the case of ArduRover instead of GUIDED it is sending GUID, basically just 4 characters of the flight mode. So to avoid confusion in modes I decided to take them from internal mode strings.

Reply:
> in the case of ArduRover instead of GUIDED it is sending GUID

I would file a bug on that.

> So to avoid confusion in modes I decided to take them from internal mode strings.

Ok, let's go with that way then and see if it causes any confusion.